libretroshare/src/rsserver/p3peers.cc

```diff
@@ -1099,6 +1109,72 @@ bool p3Peers::GetPGPBase64StringAndCheckSum(	const RsPgpId& gpg_id,
 	return true ;
 }
 
+bool p3Peers::acceptInvite( const std::string& invite,
+                            ServicePermissionFlags flags )
+{
+	if(invite.empty()) return false;
+
+	const std::string* radixPtr(&invite);
+
+	RsUrl url(invite);
```

Comment:
While passing an URL or cert works, i'm wondering if it just works by accident.
See the following from RsUrl:
When using a certificate (not an URL) `size_t schemeEndI = urlStr.find(schemeSeparator);` will return `string::npos`. This results in `mScheme = urlStr.substr(0, schemeEndI);` being the whole string ("*A value of string::npos indicates all characters until the end of the string.*") while `size_t hostBeginI = schemeEndI + 3;` will be 2.
The rest of the parsing will behave similar; using `string::npos` again `size_t queryBeginI = urlStr.find(querySeparator, hostBeginI);` (`hostBeginI` is still **2**) again for math `string qStr = urlStr.substr(queryBeginI+1, queryEndI-queryBeginI-1);`
So "parsing" a certificate with RsUrl seems to be completly broken (as it is not an url) and only happened to work by accident. It this intentional? If so maybe add a comment.

Reply:
RsUrl is a very naive but safe implementation of URL parsing, as URI standard is very complex RsUrl has no way to determine if an URL parsing has been successful or not. The acceptInvite code take in account of RsUrl being so naive so it works by design not by chance, if the parsed stuff has no `radix` in the query part (that is the case when a plain cert is passed) `radixPtr` is untouched.

Reply:
:+1: